HW0/P4/P4.py

```diff
@@ -30,47 +30,119 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    x_coords, y_coords, z_coords = np.loadtxt('P4_trajectory.txt', 
+        delimiter=',', usecols=range(3), unpack=True)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+
+    ax.plot(x_coords, y_coords, z_coords,
+            '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    Cshort = np.asarray([[1/rx,0,0],[0,1/ry,0],[0,0,1/rz]])
+    X = np.loadtxt('P4_measurements.txt', 
+        delimiter=',', usecols=range(3), unpack=True)
+    X = Cshort.dot(X)
+    x_coords = X[0,:]
+    y_coords = X[1,:]
+    z_coords = X[2,:]
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.asmatrix([
+        [1,0,0,dt,0,0],
+        [0,1,0,0,dt,0],
+        [0,0,1,0,0,dt],
+        [0,0,0,1-c*dt,0,0],
+        [0,0,0,0,1-c*dt,0],
+        [0,0,0,0,0,1-c*dt]
+        ])
+    a = np.transpose(np.asmatrix([0,0,0,0,0,g*dt]))
 
     # Initial conditions for s0
+    s = np.transpose(np.asmatrix([0, 0, 2, 15, 3.5, 4.0]))
```

Comment:
I generally like using np.diag() and then using hstack or vstack as needed. That being said, this does make it very clear what the matrix is. It's a just a bit harder to scale (imagine if it were 9 by 9).